nifi-mock/src/main/java/org/apache/nifi/util/MockProcessSession.java

```diff
@@ -793,13 +798,21 @@ private void validateState(final FlowFile flowFile) {
             throw new FlowFileHandlingException(flowFile + " is not the most recent version of this flow file within this session");
         }
 
+
         for (final List<MockFlowFile> flowFiles : transferMap.values()) {
             if (flowFiles.contains(flowFile)) {
                 throw new IllegalStateException(flowFile + " has already been transferred");
             }
         }
     }
 
+    private void validateDestinationRelation(final Relationship relationship) {
+        if (!relationships.contains(relationship)) {
+            rollback();
```

Comment:
I think, though it's for testing purpose , we should emulate as real as we can and rollback is the best possible way to revert back to a stable state. And, in a real context we may face such issue in case of orchestration, where an attribute in a flowfile may finally dictate where it should be transferred.